asset_pipeline/client.py

```diff
@@ -0,0 +1,265 @@
+import os
+import sys
+import pickle
+import urllib
+
+from oauthlib.oauth2 import LegacyApplicationClient
+from requests_oauthlib import OAuth2Session
+from oauthlib.oauth2.rfc6749.errors import InvalidGrantError
+
+from logger import logger
+from oauthlib_extras.oauth2 import MailApplicationClient
+
+# needed for insecure oauthlib http communication
+# oauth2 is, as per specification, only allowed in
+# combination with secure https communication
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
+
+APPLICATION_STATE_FILE = 'state'
+
+
+class ClientConfigParser():
+    def __init__(self, config):
+        self.config = config
+        self.protocol = config.get('protocol', None) or 'http'
+        self.host = config.get('host')
+        self.port = config.get('port')
+        self.ssl = config.get('ssl')
+        # only if we didn't specify protocol explicitly
+        if self.ssl and not config.get('protocpl'):
+            self.protocol += 's'
+        self.base_url = '{protocol}://{host}:{port}/'.format(protocol=self.protocol, host=self.host, port=self.port)
+        self.client_id = config.get('client_id')
+        self.client_secret = config.get('client_secret')
+        self.oauth_secret = config.get('oauth_secret')
```

Comment:
renamte to `auth_code` to be consistent.

Reply:
done